File: feedback-backend/app/Models/Feedback.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Feedback extends Model
{
    use HasFactory;

    protected $table = 'feedback';
    protected $primaryKey = 'feedback_id';
    public $incrementing = false;
    protected $keyType = 'string';

    protected $fillable = [
        'title',
        'description',
        'category',
        'priority',
        'status',
        'created_by',
        'assigned_to',
        'labels',
    ];

    protected $casts = [
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
    ];

    /**
     * Custom accessor for labels to handle PostgreSQL array format
     */
    public function getLabelsAttribute($value)
    {
        if (is_null($value)) {
            return [];
        }
        
        // If it's already an array, return it
        if (is_array($value)) {
            return $value;
        }
        
        // If it's a PostgreSQL array string like "{value1,value2}" or '{"value1","value2"}', parse it
        if (is_string($value)) {
            // Remove surrounding braces
            $value = trim($value, '{}');
            if ($value === '') {
                return [];
            }
            
            // Handle quoted values
            $result = [];
            $current = '';
            $inQuotes = false;
            $escaped = false;
            
            for ($i = 0; $i < strlen($value); $i++) {
                $char = $value[$i];
                
                if ($escaped) {
                    $current .= $char;
                    $escaped = false;
                    continue;
                }
                
                if ($char === '\\') {
                    $escaped = true;
                    $current .= $char;
                    continue;
                }
                
                if ($char === '"') {
                    $inQuotes = !$inQuotes;
                    continue;
                }
                
                if ($char === ',' && !$inQuotes) {
                    if ($current !== '') {
                        $result[] = $current;
                        $current = '';
                    }
                    continue;
                }
                
                $current .= $char;
            }
            
            if ($current !== '') {
                $result[] = $current;
            }
            
            return $result;
        }
        
        // Try JSON decode as fallback
        $decoded = json_decode($value, true);
        return is_array($decoded) ? $decoded : [];
    }

    /**
     * Custom mutator for labels to convert array to PostgreSQL array format
     */
    public function setLabelsAttribute($value)
    {
        if (is_null($value) || (is_array($value) && empty($value))) {
            $this->attributes['labels'] = '{}';
            return;
        }
        
        if (is_array($value)) {
            // Convert array to PostgreSQL array format: {"value1","value2"}
            $formatted = array_map(function($item) {
                // Escape backslashes and double quotes
                $item = str_replace('\\', '\\\\', (string)$item);
                $item = str_replace('"', '\\"', $item);
                return '"' . $item . '"';
            }, $value);
            
            $this->attributes['labels'] = '{' . implode(',', $formatted) . '}';
            return;
        }
        
        $this->attributes['labels'] = $value;
    }

    public function messages(): HasMany
    {
        return $this->hasMany(FeedbackMessage::class, 'feedback_id', 'feedback_id');
    }

    public function attachments(): HasMany
    {
        return $this->hasMany(FeedbackAttachment::class, 'feedback_id', 'feedback_id');
    }

    public function auditLogs(): HasMany
    {
        return $this->hasMany(FeedbackAudit::class, 'feedback_id', 'feedback_id');
    }

    public function nlpAnalysis(): HasMany
    {
        return $this->hasMany(NlpAnalysis::class, 'feedback_id', 'feedback_id');
    }
}

File: mobile-app/lib/models/feedback.dart
class Feedback {
  final String feedbackId;
  final String title;
  final String description;
  final String category;
  final String priority;
  final String status;
  final int createdBy;
  final int? assignedTo;
  final List<String> labels;
  final DateTime createdAt;
  final DateTime updatedAt;
  final List<FeedbackMessage>? messages;
  final List<FeedbackAttachment>? attachments;

  Feedback({
    required this.feedbackId,
    required this.title,
    required this.description,
    required this.category,
    required this.priority,
    required this.status,
    required this.createdBy,
    this.assignedTo,
    required this.labels,
    required this.createdAt,
    required this.updatedAt,
    this.messages,
    this.attachments,
  });

  factory Feedback.fromJson(Map<String, dynamic> json) {
    return Feedback(
      feedbackId: json['feedback_id'] ?? '',
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      category: json['category'] ?? '',
      priority: json['priority'] ?? 'MEDIUM',
      status: json['status'] ?? 'OPEN',
      createdBy: json['created_by'] ?? 0,
      assignedTo: json['assigned_to'],
      labels: List<String>.from(json['labels'] ?? []),
      createdAt: DateTime.parse(json['created_at'] ?? DateTime.now().toIso8601String()),
      updatedAt: DateTime.parse(json['updated_at'] ?? DateTime.now().toIso8601String()),
      messages: json['messages'] != null
          ? (json['messages'] as List).map((m) => FeedbackMessage.fromJson(m)).toList()
          : null,
      attachments: json['attachments'] != null
          ? (json['attachments'] as List).map((a) => FeedbackAttachment.fromJson(a)).toList()
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'description': description,
      'category': category,
      'priority': priority,
      'labels': labels,
    };
  }
}

class FeedbackMessage {
  final String messageId;
  final String feedbackId;
  final int userId;
  final String content;
  final DateTime createdAt;
  final List<FeedbackAttachment>? attachments;

  FeedbackMessage({
    required this.messageId,
    required this.feedbackId,
    required this.userId,
    required this.content,
    required this.createdAt,
    this.attachments,
  });

  factory FeedbackMessage.fromJson(Map<String, dynamic> json) {
    return FeedbackMessage(
      messageId: json['message_id'] ?? '',
      feedbackId: json['feedback_id'] ?? '',
      userId: json['user_id'] ?? 0,
      content: json['content'] ?? '',
      createdAt: DateTime.parse(json['created_at'] ?? DateTime.now().toIso8601String()),
      attachments: json['attachments'] != null
          ? (json['attachments'] as List).map((a) => FeedbackAttachment.fromJson(a)).toList()
          : null,
    );
  }
}

class FeedbackAttachment {
  final String attachmentId;
  final String feedbackId;
  final String? messageId;
  final String s3Key;
  final String fileName;
  final int? fileSize;
  final String? mimeType;
  final int uploadedBy;
  final DateTime createdAt;
  final String? downloadUrl;

  FeedbackAttachment({
    required this.attachmentId,
    required this.feedbackId,
    this.messageId,
    required this.s3Key,
    required this.fileName,
    this.fileSize,
    this.mimeType,
    required this.uploadedBy,
    required this.createdAt,
    this.downloadUrl,
  });

  factory FeedbackAttachment.fromJson(Map<String, dynamic> json) {
    return FeedbackAttachment(
      attachmentId: json['attachment_id'] ?? '',
      feedbackId: json['feedback_id'] ?? '',
      messageId: json['message_id'],
      s3Key: json['s3_key'] ?? '',
      fileName: json['file_name'] ?? '',
      fileSize: json['file_size'],
      mimeType: json['mime_type'],
      uploadedBy: json['uploaded_by'] ?? 0,
      createdAt: DateTime.parse(json['created_at'] ?? DateTime.now().toIso8601String()),
      downloadUrl: json['download_url'],
    );
  }
}

